### ISSUE TITLE
Need to add anti-overfitting factors

### ISSUE THREAD

**Issue by Applied-ML** [Applied-ML/Algo_FInance#1]
TCNs overfit very easily. Need to make sure that there exists a balance between training and validation accuracies when using SA or PSO algorithm to tune hyperparameters.
The learning rate in XGboost accounts for more than 80% of the performance variability. other than containing it on its own when using simulated annealing to tune, making sure that the highest learning rate isn't necessarily chosen just because the accuracy is better is important (account for overfitting).
